hack/generate-manifests.sh

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env bash
+
+set -eo pipefail
+set -x
+
+SRCROOT="$( CDPATH='' cd -- "$(dirname "$0")/.." && pwd -P )"
+
+KUSTOMIZE=${KUSTOMIZE:-kustomize}
+
+TEMPFILE=$(mktemp /tmp/appset-manifests.XXXXXX)
+
+if [ "$CONTAINER_REGISTRY" != "" ]; then
+	CONTAINER_REGISTRY="${CONTAINER_REGISTRY}/"
+fi
+
+
+IMAGE_NAMESPACE="${IMAGE_NAMESPACE:-argoprojlabs}"
+IMAGE_TAG="${IMAGE_TAG:-}"
+
+# if the tag has not been declared, and we are on a release branch, use the VERSION file.
+if [ "$IMAGE_TAG" = "" ]; then
+  branch=$(git rev-parse --abbrev-ref HEAD || true)
+  if [[ $branch = release-* ]]; then
+    pwd
+    IMAGE_TAG=v$(cat $SRCROOT/VERSION)
+  fi
+fi
+# otherwise, use latest
+if [ "$IMAGE_TAG" = "" ]; then
+  IMAGE_TAG=latest
+fi
+
```

Comment:
We should add header comment "# This is an auto-generated file. DO NOT EDIT" to generated manifest to warn people.   Also, we will need CI validation to fail jobs that commit mismatch manifests (i.e. differ from generated manifests)

Reply:
@wtam2018  Done and done.